src/components/backend-ai-storage-list.ts

```diff
@@ -1266,6 +1270,22 @@ export default class BackendAiStorageList extends BackendAIPage {
     this.shadowRoot.querySelector('#fileInput').value = '';
   }
 
+  runFileUploadQueue(session = null) {
+    if (session !== null) {
+      (this.fileUploadQueue as any).push(session);
+    }
+    if (this.fileUploadCount < this.concurrentFileUploadLimit) {
+      let queuedSession;
+      for (let i = this.fileUploadCount; i < this.concurrentFileUploadLimit + 1; i++) {
+        if (this.fileUploadQueue.length > 0) {
+          queuedSession = this.fileUploadQueue.shift();
+          this.fileUploadCount = this.fileUploadCount + 1;
+          queuedSession.start();
+        }
+      }
+    }
```

Comment:
I think this works well. Just a minor question that I was unable to understand.

* `if` statement in line 1277 needed? `for` loop in two lines below anyway will not run if file upload count is larger than concurrent upload limit.
* Reason of `+ 1` in line 1279? It seems like three files can concurrently be uploaded if `concurrentFileUploadLimit` is set to 2.

Reply:
1. If 2 files are already uploading, code 1278-1285 does not run. If one or no file is uploading now, the codes will run to start file sessions. So it is intentional.
2. Bug. I will fix it.

Reply:
@adrysn Fixed the second one.

Reply:
@adrysn  Fixed the first one.